### ISSUE TITLE
Usage on alpine docker image

### ISSUE THREAD

**Issue by hubby2004** [citrix/terraform-provider-citrixadc#116]
**Describe the bug**
Running the terraform-provider-citrixadc does not work in an alpine docker image like the one shipped by [Hashicorp for Terraform](https://hub.docker.com/r/hashicorp/terraform/) 

**To Reproduce**
* Run a container based on `hashicorp/terraform:light image`
* Manually install the terraform-provider-citrixadc provider under .terraform/plugins folder
* Run `terraform init`
* Run `terraform validate`
The failure occurs:

```
Failed to instantiate provider "registry.terraform.io/hashicorp/citrixadc" to obtain schema: fork/exec .terraform/plugins/registry.terraform.io/hashicorp/citrixadc/0.12.34/linux_amd64/terraform-provider-citrixadc_v0.12.34: no such file or directory
```
**Expected behavior**
Terraform validate should run without any issues.

**Logs**
Below is an example of what the pipeline output is when running the `terraform validate` after downloading the provider and placing it in `.terraform/plugins` folder
```
$ terraform init
Initializing modules...
Initializing the backend...
Initializing provider plugins...
- Using previously-installed hashicorp/citrixadc v0.12.34
- Using previously-installed hashicorp/vsphere v1.24.0
The following providers do not have any version constraints in configuration,
so the latest version was installed.
To prevent automatic upgrades to new major versions that may contain breaking
changes, we recommend adding version constraints in a required_providers block
in your configuration, with the constraint strings suggested below.
* hashicorp/citrixadc: version = "~> 0.12.34"
* hashicorp/vsphere: version = "~> 1.24.0"
Terraform has been successfully initialized!
You may now begin working with Terraform. Try running "terraform plan" to see
any changes that are required for your infrastructure. All Terraform commands
should now work.
If you ever set or change modules or backend configuration for Terraform,
rerun this command to reinitialize your working directory. If you forget, other
commands will detect it and remind you to do so if necessary.
$ terraform providers
Providers required by configuration:
.
├── provider[registry.terraform.io/hashicorp/citrixadc]
├── provider[registry.terraform.io/hashicorp/vsphere]
└── module.dsc_server
    └── provider[registry.terraform.io/hashicorp/vsphere]
Providers required by state:
    provider[registry.terraform.io/hashicorp/vsphere]
$ wget https://github.com/citrix/terraform-provider-citrixadc/releases/download/v0.12.34/terraform-provider-citrixadc_0.12.34_linux_amd64.tar.gz
Connecting to github.com (140.82.112.4:443)
Connecting to github-production-release-asset-2e65be.s3.amazonaws.com (52.217.13.228:443)
saving to 'terraform-provider-citrixadc_0.12.34_linux_amd64.tar.gz'
terraform-provider-c 100% |********************************| 8657k  0:00:00 ETA
'terraform-provider-citrixadc_0.12.34_linux_amd64.tar.gz' saved
$ tar -zxf terraform-provider-citrixadc_0.12.34_linux_amd64.tar.gz
$ rm README.md LICENSE terraform-provider-citrixadc_0.12.34_linux_amd64.tar.gz
$ cp terraform-provider-citrixadc .terraform/plugins/registry.terraform.io/hashicorp/citrixadc/0.12.34/linux_amd64/terraform-provider-citrixadc_v0.12.34
$ chmod 755 .terraform/plugins/registry.terraform.io/hashicorp/citrixadc/0.12.34/linux_amd64/terraform-provider-citrixadc_v0.12.34
$ terraform validate
Error: Could not load plugin
Plugin reinitialization required. Please run "terraform init".
Plugins are external binaries that Terraform uses to access and manipulate
resources. The configuration provided requires plugins which can't be located,
don't satisfy the version constraints, or are otherwise incompatible.
Terraform automatically discovers provider requirements from your
configuration, including providers used in child modules. To see the
requirements and constraints, run "terraform providers".
Failed to instantiate provider "registry.terraform.io/hashicorp/citrixadc" to
obtain schema: fork/exec
.terraform/plugins/registry.terraform.io/hashicorp/citrixadc/0.12.34/linux_amd64/terraform-provider-citrixadc_v0.12.34:
no such file or directory
ERROR: Job failed: exit code 1
```

**Environment (please complete the following information):**
 - OS: CentOS 8
 - Provider version [e.g. chrome, safari]
 - Terraform version 0.13.3
 - Citrix ADC version 12.1

**Additional context**
The issue happens in the pipeline I am running to configure lbservers in a VPX. This is important for us as we are moving to infrastructure as code. Having this provider working in an alpine-based image is important for anyone using docker containers to run pipelines.

I am not a go programmer but I believe this can be achieved by adding the following 👍 

```
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build
```

**Comment by hubby2004** [comment]
When I rebuilt the provider with `CGO_ENABLED=0 GOOS=linux GOARCH=amd64`, I was able to load it fine as a provider in Terraform container.

**Comment by giorgos-nikolopoulos** [comment]
The solution to that would be to modify the goreleaser configuration to put in effect the required variables when building the binaries.

This way the binaries provided under releases will have the correct options set.

Will take this up when time permits.

**Comment by giorgos-nikolopoulos** [comment]
Fixed with #135

From release 0.12.39 produced binaries from goreleaser have the flag set `CGO_ENABLED=0`
